src/app/clusters/localization-time-format-server/localization-time-format-server.cpp

```diff
@@ -0,0 +1,150 @@
+    CHIP_ERROR ReadSupportedCalendarTypes(AttributeValueEncoder & aEncoder);
+};
+
+LocalizationTimeFormatAttrAccess gAttrAccess;
+
+CHIP_ERROR LocalizationTimeFormatAttrAccess::ReadIfSupported(CHIP_ERROR (ConfigurationManager::*getter)(uint8_t &),
+                                                             AttributeValueEncoder & aEncoder)
+{
+    uint8_t data;
+    CHIP_ERROR err = (DeviceLayer::ConfigurationMgr().*getter)(data);
+
+    if (err == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+    {
+        data = 0;
+    }
+    else if (err != CHIP_NO_ERROR)
+    {
+        return err;
+    }
+
+    return aEncoder.Encode(data);
+}
+
+CHIP_ERROR LocalizationTimeFormatAttrAccess::WriteIfSupported(CHIP_ERROR (ConfigurationManager::*setter)(uint8_t),
+                                                              AttributeValueDecoder & aDecoder)
+{
+    uint8_t data;
+
+    ReturnErrorOnFailure(aDecoder.Decode(data));
+
+    return (DeviceLayer::ConfigurationMgr().*setter)(data);
+}
+
+CHIP_ERROR LocalizationTimeFormatAttrAccess::ReadSupportedCalendarTypes(AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    DeviceLayer::AttributeList<LocalizationTimeFormat::CalendarType, DeviceLayer::kMaxCalendarTypes> supportedCalendarTypes;
+
+    if (DeviceLayer::PlatformMgr().GetSupportedCalendarTypes(supportedCalendarTypes) == CHIP_NO_ERROR)
+    {
+        err = aEncoder.EncodeList([&supportedCalendarTypes](const auto & encoder) -> CHIP_ERROR {
+            for (auto type : supportedCalendarTypes)
+            {
+                ReturnErrorOnFailure(encoder.Encode(type));
+            }
+
+            return CHIP_NO_ERROR;
+        });
+    }
+    else
+    {
+        err = aEncoder.EncodeEmptyList();
+    }
+
+    return err;
+}
+
+CHIP_ERROR LocalizationTimeFormatAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == LocalizationTimeFormat::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case LocalizationTimeFormat::Attributes::HourFormat::Id:
+        return ReadIfSupported(&ConfigurationManager::GetHourFormat, aEncoder);
+    case LocalizationTimeFormat::Attributes::CalendarType::Id:
+        return ReadIfSupported(&ConfigurationManager::GetCalendarType, aEncoder);
+    case LocalizationTimeFormat::Attributes::SupportedCalendarTypes::Id:
+        return ReadSupportedCalendarTypes(aEncoder);
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR LocalizationTimeFormatAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    VerifyOrDie(aPath.mClusterId == LocalizationTimeFormat::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case LocalizationTimeFormat::Attributes::HourFormat::Id:
+        return WriteIfSupported(&ConfigurationManager::StoreHourFormat, aDecoder);
+    case LocalizationTimeFormat::Attributes::CalendarType::Id:
```

Comment:
Why are we adding a bunch of logic for this, with more per-platform logic needing to be written to make this work, instead of just storing them in NVM and having this work with a lot less code on all platforms?  This does not seem like the right approach to me.